02_Util/src/util/directus.ts

```diff
@@ -0,0 +1,245 @@
+// Copyright Contributors to the CitrineOS Project
+//
+// SPDX-License-Identifier: Apache 2.0
+
+import { SystemConfig } from "@citrineos/base";
+import { authentication, DirectusFlow, DirectusOperation, RestClient, createDirectus, createFlow, createOperation, readFlows, rest, staticToken, updateFlow, updateOperation } from "@directus/sdk";
+import { RouteOptions } from "fastify";
+import { JSONSchemaFaker } from "json-schema-faker";
+import { Logger, ILogObj } from "tslog";
+
+interface Schema {
+    // No custom collections needed
+}
+
+export class DirectusUtil {
+
+    protected readonly _config: SystemConfig;
+    protected readonly _logger: Logger<ILogObj>;
+    private readonly _client: RestClient<Schema>;
+
+    constructor(config: SystemConfig, logger?: Logger<ILogObj>) {
+        this._config = config;
+        this._logger = logger ? logger.getSubLogger({ name: this.constructor.name }) : new Logger<ILogObj>({ name: this.constructor.name });
+        let client;
+        if (this._config.util.directus?.token) { // Auth with static token
+            client = createDirectus(`http://${this._config.util.directus?.host}:${this._config.util.directus?.port}`)
+                .with(staticToken(this._config.util.directus?.token)).with(rest());
+        } else if (this._config.util.directus?.username && this._config.util.directus?.password) { // Auth with username and password
+            client = createDirectus<Schema>(`http://${this._config.util.directus?.host}:${this._config.util.directus?.port}`)
+                .with(authentication()).with(rest());
+            this._logger.info(`Logging into Directus as ${this._config.util.directus.username}`);
+            client.login(this._config.util.directus.username, this._config.util.directus.password);
+        } else { // No auth
```

Comment:
Can this actually happen? Since we set defaults in the schema above. I don't think there is a way to clear them.

Reply:
Right now, these defaults don't apply; there's an issue with Zod regarding how default works with nested optional objects. This is a good point though--there shouldn't be a default username/password. I'll remove it!